src/KubernetesClient/Informers/KubernetesInformer.cs

```diff
@@ -0,0 +1,155 @@
+                            cancellationToken: cancellationToken).ConfigureAwait(false);
+                        if (!response.Response.IsSuccessStatusCode)
+                            throw new HttpRequestWithStatusException("Web server replied with error code") {StatusCode = response.Response.StatusCode};
+
+                        var listKubernetesObject = response.Body;
+                        _resourceVersion = listKubernetesObject.Metadata.ResourceVersion;
+                        var items = listKubernetesObject.Items ?? new List<TResource>();
+                        foreach (var item in items.ToReset(_type == ResourceStreamType.ListWatch))
+                        {
+                            if (cancellationToken.IsCancellationRequested)
+                                break;
+                            observer.OnNext(item);
+                        }
+                    })
+                    .WithRetryPolicy(_parent._retryPolicy);
+            }
+
+            private IObservable<ResourceEvent<TResource>> Watch()
+            {
+                return Observable.Create<ResourceEvent<TResource>>(async (observer, cancellationToken) =>
+                    {
+                        var result = await _parent._kubernetes.ListWithHttpMessagesAsync<TResource>(
+                            watch: true,
+                            allowWatchBookmarks: true,
+                            resourceVersion: _resourceVersion,
+                            timeoutSeconds: int.MaxValue,
+                            cancellationToken: cancellationToken
+                        ).ConfigureAwait(false);
+                        if (!result.Response.IsSuccessStatusCode)
+                            throw new HttpRequestWithStatusException("Web server replied with error code") {StatusCode = result.Response.StatusCode};
+                        return Task.FromResult(result)
+                            .Watch()
+                            .SelectMany(x => // this is not a one to one mapping as some events cause side effects but don't propagate, so we need SelectMany
+                            {
+                                switch (x.Type)
+                                {
+                                    case WatchEventType.Added:
+                                        return new[] {x.Object.ToResourceEvent(EventTypeFlags.Add)};
+                                    case WatchEventType.Deleted:
+                                        return new[] {x.Object.ToResourceEvent(EventTypeFlags.Delete)};
+                                    case WatchEventType.Modified:
+                                        return new[] {x.Object.ToResourceEvent(EventTypeFlags.Modify)};
+                                    case WatchEventType.Bookmark:
+                                        if (x.Object is IMetadata<V1ObjectMeta> status)
+                                            _resourceVersion = status.Metadata.ResourceVersion;
```

Comment:
I'm not sure that bookmarks can be solely replied upon to keep `_resourceVersion` up to date. The docs state:
>  Clients should not assume bookmarks are returned at any specific interval, nor may they assume the server will send any BOOKMARK event during a session.

We should use the same approach as the Go [client](https://github.com/kubernetes/client-go/blob/2a5c88adcc5c656a3e843a3f2ca4633cdb9b4227/tools/cache/reflector.go#L474) and update `_resourceVersion` on every `{Added, Deleted, Modified, Bookmark}` event.

Reply:
So this part has me really confused and I couldn't find a straight forward answer. Conceptually we have resourceVersion on LIST object and each item in it. When you execute a List or Watch, you're actually supplying a resourceVersion of the LIST. The relationship between list version and individual item versions is not described anywhere. For example if they were implemented as a simple integer increments, then LIST version would have completely different semantics from items in it, and its own counter. On the other hand if we're saying that it's just a timestamp of latest mutation of the list, then obtaining the latest "version" from a Add/Deleted/Modified resource can be applied to List version. I could not find any documentation in K8S to clarify if this assumption is safe to make given that they state resourceVersion should be treated "opaque". If go does this, then I'll apply same behavior, I just wanted someone to clarify this for me.

Reply:
I've changed it to take a resource version from any watch event that has it set, not just bookmark. Double-check please that this is what we're expecting.